tigramite/independence_tests/independence_tests_base.py

```diff
@@ -545,12 +546,8 @@ def run_test_raw(self, x, y, z=None, x_type=None, y_type=None, z_type=None, alph
 
 
         # Get the p-value (returns None if significance='fixed_thres')
-        if has_data_type:
-            pval = self._get_p_value(val=val, array=array, xyz=xyz,
-                    T=T, dim=dim, data_type=data_type)
-        else:
-            pval = self._get_p_value(val=val, array=array, xyz=xyz,
-                    T=T, dim=dim)
+        pval = self._get_p_value(val=val, array=array, xyz=xyz,
```

Comment:
data_type may not be defined, so this would throw an error...

Reply:
in _get_p_value, i added the following lines:

` elif use_sig == 'shuffle_test':
            if data_type is None:
                pval = self.get_shuffle_significance(array=array,
                                                     xyz=xyz,
                                                     value=val)
            else:
                pval = self.get_shuffle_significance(array=array,
                                                     xyz=xyz,
                                                     value=val,
                                                     data_type=data_type)`

so if it data_type is none, would this still throw errors?

Reply:
yes, because it doesn't exist and cannot be set as an argument

Another thing: better merge into developer!